lib/client/api_matrix.es6.js

```diff
@@ -53,6 +53,110 @@ foam.CLASS({
+          return browsers;
+        });
+      },
+    },
+    {
+      name: 'filterMatrix_',
+      documentation: `A post process of toMatrix, the result api toMatrix
+        will be filtered based on options.browserOptions and options.
+        numAvailable.`,
+      args: [
+        {
+          name: 'matrix',
+          typeName: 'JSON',
+          documentation: `JSON of the form:
+            {interfaceName: {apiName: {BrowserKey: true, ...} ...} ... } ...}`,
+        },
+        {
+          name: 'options',
+          typeName: 'JSON',
+          documentation: `An optional argument are used to filter the
+            matrix result:
+              searchKey: the result interface/API must contains this String
+                value, ignore cases.
+              browserOptions: An JSON of form {browserKey: boolean, ...}.
+                The returned result are filtered. So that only APIs that
+                whether it is available in this browser equals to
+                options[browser] are returned.
+                This option can be used to do queries such as finding
+                proprietary apis for a browser in a set of browsers.
+              numAvailable: an interger or interger Array that the web
+                interface whose number of available browser not included
+                in this array or not equal to this interger are filtered.
+                This option can be used to do queries such as finding apis
+                that exitsin at lease 3 browser vendors.`,
+        },
+      ],
+      returns: {
+        typeName: 'JSON',
+        documentation: `JSON of the form:
+            {interfaceName: {apiName: {BrowserKey: true, ...} ...} ... } ...}
+            This structure is easy for displaying a nested table.`,
+      },
+      code: function(matrix, options) {
+        options = options || {};
+        let browserOptions = options.browserOptions;
+        let numAvailable = options.numAvailable;
+        if (!browserOptions && !numAvailable) return matrix;
+        for (let webInterface in matrix) {
+          if (!matrix.hasOwnProperty(webInterface)) continue;
+          for (let api in matrix[webInterface]) {
+            if (!Object.prototype.hasOwnProperty
+              .call(matrix[webInterface], api)) continue;
+            if (numAvailable) {
+              if ((!Array.isArray(numAvailable) && numAvailable !==
+                Object.keys(matrix[webInterface][api]).length) ||
+                numAvailable.indexOf(Object.keys(matrix[webInterface][api])
+                .length) === -1) {
+                  delete matrix[webInterface][api];
+                  continue;
+                }
+            }
+            if (browserOptions) {
+              for(let browserKey in browserOptions) {
+                if (!browserOptions.hasOwnProperty(browserKey)) continue;
+                if (matrix[webInterface][api].hasOwnProperty(browserKey) !==
+                  browserOptions[browserKey]) {
```

Comment:
This indentation feels backwards. I think the contents of the scope should be +2 from the parent, and the continuation should be +4. I really need to read the JS style guide, but that's what I'd expect in any other language.